src/libstd/sys/unix/stack_overflow.rs

```diff
@@ -71,7 +75,39 @@ mod imp {
         (*(info as *const siginfo_t)).si_addr as usize
     }
 
-    #[cfg(not(any(target_os = "linux", target_os = "android")))]
+    #[cfg(all(target_os = "netbsd", target_pointer_width = "64"))]
+    unsafe fn siginfo_si_addr(info: *mut libc::siginfo_t) -> usize {
+        #[repr(C)]
+        struct siginfo_t {
+            a: [libc::c_int; 4], // si_signo, si_code, si_errno, pad to 8 byte boundary
+            si_addr: *mut libc::c_void,
+        }
+
+        (*(info as *const siginfo_t)).si_addr as usize
+    }
+
+    #[cfg(any(target_os = "freebsd",
+              target_os = "dragonfly"))]
+    unsafe fn siginfo_si_addr(info: *mut libc::siginfo_t) -> usize {
+        #[repr(C)]
+        struct siginfo_t {
+            a: [libc::c_int; 3], // si_signo, si_code, si_errno
+            b: libc::pid_t,      // si_pid
+            c: libc::uid_t,      // si_gid
+            d: libc::c_int,      // si_status
+            si_addr: *mut libc::c_void,
+        }
+
+        (*(info as *const siginfo_t)).si_addr as usize
```

Comment:
These typically arise because `siginfo_t` is a union in C which we can't express well in libc, so we end up doing something like:

```rust
pub struct siginfo_t {
    pub si_signo: ::c_int,
    pub si_errno: ::c_int,
    pub si_code: ::c_int,
    pub _pad: [::c_int; 29],
}
```

where it's up to you to cast `&info._pad` to whatever struct you want out of the union.

@dhuseby it looks like freebsd/dragonfly  already have `si_addr` in this structure, though?

```rust
pub struct siginfo_t {
    pub si_signo: ::c_int,
    pub si_errno: ::c_int,
    pub si_code: ::c_int,
    pub si_pid: ::pid_t,
    pub si_uid: ::uid_t,
    pub si_status: ::c_int,
    pub si_addr: *mut ::c_void,
    _pad: [::c_int; 12],
}
```